### PR TITLE
5.x merge 4.x

### PR DESCRIPTION
**Pull request by asmorkalov**
The PR includes all changes in 4.x till 4.9.0 release.

OpenCV Contrib PR: https://github.com/opencv/opencv_contrib/pull/3621
OpenCV Extra PR: https://github.com/opencv/opencv_extra/pull/1142
Requires CI update in: https://github.com/opencv/ci-gha-workflow/pull/134

#23021 from cudawarped:add_cuda_first_class_support
#23654 from zihaomu:wino_fp16
#24060 from TolyaTalamanov:at/advanced-device-selection-onnxrt-directml
#24068 from TolyaTalamanov:at/add-onnx-coreml-execution-provider
#24136 from komakai:visionos_support
#24205 from PeterJohnson:fix-msvc-arm64
#24224 from AsyaPronina:asyadev/port_vas_ot_to_opencv
#24243 from kecsap:4.x
#24271 from Kumataro:fix24163
#24299 from dkurt:qrcode_decode
#24364 from bagelbytes61:bugfix/qrcode-version-estimator
#24378 from fengyuentau:instance_norm
#24384 from Dhanwanth1803:feat-crop
#24396 from Tsai-chia-hsiang:yolov8cv
#24397 from richard28039:add_fcnresnet101_to_dnn_sample
#24445 from Abdurrahheem:ash/dev_einsum_pref
#24456 from alexlyulkov:al/aar
#24458 from laolaolulu:4.x
#24462 from fengyuentau:cann_support_hardswish
#24463 from dkurt:dnn_shared_nodes_fusion
#24465 from ivashmak:fix_usac_tutorial
#24466 WanliZhong:refactor_softmax
#24473 from alexlyulkov:al/samples_with_maven
#24474 from asmorkalov:as/BOWImgDescriptorExtractoor_java_ctor
#24476 from fengyuentau:attention_layer
#24477 from asmorkalov:as/SimpleBlobDetector_js
#24479 from AleksandrPanov:fix_contour_filtering
#24480 from savuor:backport_patch_nans
#24483 from dkurt:dnn_fusion_commutative_ops
#24485 from hanliutong:rvv-opt
#24488 from hipudding:npu_support
#24492 from asmorkalov:as/reprioritize_fps_source_ffmpeg
#24495 from vrabaud:fast_math_compile
#24496 from cabelo:yolov3
#24499 from ivashmak:usac_bug_fix
#24500 from fengyuentau:test_layer_fusion
#24509 from Abdurrahheem:ash/dev_einsum_fast_gemm
#24510 from asmorkalov:as/softmax_rvv
#24511 from asmorkalov:as/remove_rvv_internals_from_docs
#24512 from asmorkalov:as/gstreamer_warn_fix
#24519 from shubh1176:4.x
#24521 from dkurt:fix_broken_urls
#24526 from thewoz:arrow
#24527 from MaximSmolskiy:fix-out-of-image-corners-in-cv-cornersubpix
#24531 from fengyuentau:instancenorm_onnx_test_cuda
#24533 from AleksandrPanov:aruco_remove_filter_fix
#24535 from dkurt:ipp_distransform_update
#24539 from LaurentBerger:blobrecttoimage
#24544 from fengyuentau:layernorm_conformance
#24546 from thewoz:checkerboard
#24547 from WanliZhong:refactor_conv_perf_test
#24552 from fengyuentau:layernorm_backends
#24553 from skycat8:yolov5
#24554 from asmorkalov:as/android_sdk_local_props
#24560 from asmorkalov:as/openni_status
#24565 from CNClareChen:4.x
#24566 from MaximSmolskiy:fix-typos-in-calibinit.cpp
#24569 from Abdurrahheem:ash/padding_value_fix
#24575 from asmorkalov:as/drop_android_engine
#24576 from AsyaPronina:ot_to_python
#24577 from dkurt:dnn_graph_match_stack
#24581 from vrabaud:tsan
#24582 from hanliutong:rvv-lut
#24584 from TolyaTalamanov:at/implement-inference-only-mode-for-ov-backend
#24592 from savuor:recorder_android
#24595 from MaximSmolskiy:fix-typo-inChessBoardDetector-generateQuads
#24597 from MaximSmolskiy:fix-bug-in-ChessBoardDetector-findQuadNeighbors
#24598 from AleksandrPanov:add_android_qr_sample
#24599 from asmorkalov:as/android_face_detect_dnn
#24600 from asmorkalov:as/tutorial2_fix
#24601 from TolyaTalamanov:at/fix-onnxrt-include-path-in-cmake
#24604 from MaximSmolskiy:fix-time-measurement-in-test-Calib3d_ChessboardDetector.timing
#24605 from MaximSmolskiy:speed-up-ChessBoardDetector-findQuadNeighbors
#24607 from asmorkalov:as/serialize_android_samples_build
#24608 from spacegaier:patch-1
#24610 from jimmylaw21:dnn-onnx-add-group-norm-layer
#24611 from Abdurrahheem:ash/add_yolov6_test
#24613 from WanliZhong:softmax_default_axis
#24615 from smirnov-alexey:as/infer2_assert_soften
#24616 from asmorkalov:as/android_mobilenet_refresh
#24618 from vrabaud:compilation
#24619 from philnelson:patch-1
#24622 from alexlyulkov:al/fixed-android-sample-img-save
#24625 from asmorkalov:as/mjpeg_encoder_status
#24629 from Kumataro:add_crosscompile_with_multiarch
#24630 from tomoaki0705:fix_typo_calib3d
#24634 from jubinchheda:deprecated-ios-api-patches
#24639 from asmorkalov:as/android_ocl_rework
#24640 from asmorkalov:as/android_info_lib_cleanup
#24641 from asmorkalov:as/android_samples_drop_mk
#24642 from tomoaki0705:merge_features_aarch64
#24647 from fengyuentau:cuda_sub
#24648 from vrabaud:compilation
#24649 from asmorkalov:as/android_camera2_extact_request
#24651 from asmorkalov:as/android_camera2_resolution_report
#24652 from asmorkalov:as/opencvai
#24653 from savuor:doc_android_tutorials
#24655 from fengyuentau:graph_simplifier_optional_input
#24657 from asmorkalov:as/ffmpeg_timeout_warning
#24658 from smirnov-alexey:as/gapi_ov_get_model_layout
#24659 from MaximSmolskiy:fix-bug-in-ChessBoardDetector-cleanFoundConnectedQuads
#24662 from asmorkalov:as/android_native_camera
#24664 from tomoaki0705:warning_fix_clang
#24665 from opencv-pushbot:gitee/alalek/fix_winpack_vc14
#24666 from zzuliys:4.x
#24667 from MaximSmolskiy:fix-mismatch-and-simplify-code-in-ChessBoardDetector-findQuadNeighbors
#24668 from vrabaud:compilation
#24672 from dkurt:adjust_slice_optional_inputs
#24673 from tomoaki0705:roll_back_23654
#24680 from AleksandrPanov:update_android_mobilenet_tutorial
#24681 from WanliZhong:err_armv8
#24685 from AleksandrPanov:fix_build_grandle
#24689 from FantasqueX:update-libspng-0-7-4
#24692 from savuor:doc_android_tutorial_camera
#24693 from FantasqueX:fix-spng-homepage
#24694 from fengyuentau:matmul_refactor
#24696 from Kumataro:fix24695
#24698 from tomoaki0705:support_arm64_windows
#24700 from savuor:doc_android_win_fix
#24701 from dodo920306:4.x
#24702 from vrabaud:compilation
#24706 from Kumataro:fix24705
#24709 from vpisarev:winograd_mode
#24713 from MaximSmolskiy:improve-icvSmoothHistogram256
#24715 from AleksandrPanov:update_android_opencl_sample
#24716 from asmorkalov:as/spng_warning_fix
#24717 from FantasqueX:zlib-1-3
#24718 from asmorkalov/as/system_spng
#24723 from asmorkalov:as/android_drop_old_toolchain
#24728 from opencv-pushbot:gitee/alalek/samples_python_312
#24730 from mshabunin:fix-mali-opencl
#24732 from asmorkalov:as/android_toolchain_install
#24736 from opencv-pushbot:gitee/alalek/issue_24734
#24737 from opencv-pushbot:gitee/alalek/issue_24735
#24740 from opencv-pushbot:gitee/alalek/ocl_fix_kernel_compilation
#24741 from FantasqueX:fix-zlib-version-parse
#24744 from chacha21:cmake_cuda_help
#24745 from tomoaki0705:cmake_follow_up
#24747 from asmorkalov:as/tune_vitb_cuda
#24749 from FantasqueX:fix-openjpeg-version
#24751 from dkurt:d.kurtaev/hotfix_cuda_scale
#24752 from asmorkalov:as/android_aar_with_gradle
#24753 from fengyuentau:einsum_importer
#24754 from FantasqueX:refine-png-version-parsing
#24755 from MaximSmolskiy:fix-typos-in/imgproc/src/contours.cpp
#24759 from jubinchheda:4.x
#24761 from opencv-pushbot:gitee/alalek/test_skip_update_win32
#24763 from opencv-pushbot:gitee/alalek/test_dnn_skip_update_debug
#24764 from asmorkalov:as/python_read_only_docs
#24765 from fengyuentau:mod_operator
#24766 from asmorkalov:update_version_4.9.0-pre
#24767 from opencv-pushbot:gitee/alalek/update_ffmpeg_4.x
#24769 from asmorkalov:as/calib_doc_update
#24771 from FantasqueX:remove-is-clang
#24773 from tailsu:sd/pathlike
#24775 from opencv-pushbot:gitee/alalek/build_python2_disable_by_default
#24776 from opencv-pushbot:gitee/alalek/docs_exclude_bindings_utils
#24778 from asmorkalov:as/win_msmf_stream_type_check
#24781 from FantasqueX:remove-deprecated-cmake-variable-2
#24783 from Abdurrahheem:ash/yolov7_weights
#24788 from asmorkalov:as/android_samples_repo_typo
#24790 from FantasqueX:fix-avif-version-detect
#24793 from Kumataro:fix24792
#24796 from georgthegreat:patch-2
#24798 from Rageking8:correct-invalid-error-directive
#24799 from cudawarped:fix_24789
#24801 from theartful:better_error_msg
#24805 from MengqingCao:dvpp_support
#24807 from vrabaud:aruco_fix
#24808 from fengyuentau:fix_layernorm
#24812 from Abdurrahheem:ash/einsum_bachedGemm
#24814 from cudawarped:fix_core_clang_warnings
#24815 from brad0:openbsd_x86_build
#24819 from brad0:freebsd_aarch64
#24820 from brad0:openbsd_build
#24823 from shenlebantongying:fix-qt-statusbar
#24827 from alexlyulkov:al/java-camera-view-orientation
#24828 from cudawarped:fix_24774_cap_ms
#24829 from vrabaud:aruco_fix
#24834 from fengyuentau:cuda_naryeltwise_broadcast
#24837 from asmorkalov:as/python_limited_api_version
#24840 from fengyuentau:ocl_innerproduct
#24842 from asmorkalov:as/android_aar_docs
#24843 from asmorkalov:as/android_drop_aidl
#24846 from philnelson:patch-2
#24849 from alexlyulkov:al/aar-javadoc

Previous "Merge 4.x": #24486

```
force_builders=Linux OpenCL,Linux AVX2,Linux32,Win32
```

### CODE REVIEW

File: modules/dnn/include/opencv2/dnn/version.hpp [hunk 1]
```diff
@@ -6,7 +6,7 @@
 #define OPENCV_DNN_VERSION_HPP
 
 /// Use with major OpenCV version only.
-#define OPENCV_DNN_API_VERSION 20230620
+#define OPENCV_DNN_API_VERSION 20231225
```
Comment (asmorkalov) [inline modules/dnn/include/opencv2/dnn/version.hpp:9]: @opencv-alalek Could you tune abi-complience-checker on CI.

Comment (opencv-alalek) [reply modules/dnn/include/opencv2/dnn/version.hpp:9]: > abi-complience-checker 

That result is ignored on 5.x branch

File: modules/3d/include/opencv2/3d.hpp [hunk 2]
```diff
@@ -1560,7 +1560,7 @@ unit length.
  */
 CV_EXPORTS_W void decomposeEssentialMat( InputArray E, OutputArray R1, OutputArray R2, OutputArray t );
 
-/** @brief Recovers the relative camera rotation and the translation from corresponding points in two images from two different cameras, using chirality check. Returns the number of
+/** @brief Recovers the relative camera rotation and the translation from corresponding points in two images from two different cameras, using cheirality check. Returns the number of
```
Comment (opencv-alalek) [inline modules/3d/include/opencv2/3d.hpp:1563]: Why?
There are no such changes on 4.x: https://github.com/opencv/opencv/blame/4.9.0/modules/calib3d/include/opencv2/calib3d.hpp#L2739

File: modules/3d/include/opencv2/3d.hpp [hunk 3]
```diff
@@ -1592,11 +1592,11 @@ line in pixels, beyond which the point is considered an outlier and is not used
 final fundamental matrix. It can be set to something like 1-3, depending on the accuracy of the
 point localization, image resolution, and the image noise.
 @param mask Input/output mask for inliers in points1 and points2. If it is not empty, then it marks
-inliers in points1 and points2 for the given essential matrix E. Only these inliers will be used to
-recover pose. In the output mask only inliers which pass the chirality check.
+inliers in points1 and points2 for then given essential matrix E. Only these inliers will be used to
```
Comment (opencv-alalek) [inline modules/3d/include/opencv2/3d.hpp:1595]: wrong

File: modules/3d/include/opencv2/3d.hpp [hunk 4]
```diff
@@ -1592,11 +1592,11 @@ line in pixels, beyond which the point is considered an outlier and is not used
...
+recover pose. In the output mask only inliers which pass the cheirality check.
 
 This function decomposes an essential matrix using @ref decomposeEssentialMat and then verifies
-possible pose hypotheses by doing chirality check. The chirality check means that the
+possible pose hypotheses by doing cheirality check. The cheirality check means that the
```
Comment (opencv-alalek) [inline modules/3d/include/opencv2/3d.hpp:1599]: wrong again

List of changes on 4.x branch is here:

```
git diff 5.x-merge-4.x-origin..4.9.0 -- modules/calib3d/include/opencv2/calib3d.hpp
```

File: modules/calib/include/opencv2/calib.hpp [hunk 5]
```diff
@@ -411,7 +411,8 @@ enum { CALIB_CB_ADAPTIVE_THRESH = 1,
        CALIB_CB_EXHAUSTIVE      = 16,
        CALIB_CB_ACCURACY        = 32,
        CALIB_CB_LARGER          = 64,
-       CALIB_CB_MARKER          = 128
+       CALIB_CB_MARKER          = 128,
+       CALIB_CB_PLAIN           = 256
```
Comment (opencv-alalek) [inline modules/calib/include/opencv2/calib.hpp:415]: Missing these parts from 4.x:


```.diff
@@ -1235,6 +1236,10 @@ square-like shape) to filter out false quads extracted at the contour retrieval
 -   @ref CALIB_CB_FAST_CHECK Run a fast check on the image that looks for chessboard corners,
 and shortcut the call if none is found. This can drastically speed up the call in the
 degenerate condition when no chessboard is observed.
+-   @ref CALIB_CB_PLAIN All other flags are ignored. The input image is taken as is.
+No image processing is done to improve to find the checkerboard. This has the effect of speeding up the
+execution of the function but could lead to not recognizing the checkerboard if the image
+is not previously binarized in the appropriate manner.
 
 The function attempts to determine whether the input image is a view of the chessboard pattern and
 locate the internal chessboard corners. The function returns a non-zero value if all of the corners
@@ -1595,6 +1600,10 @@ The algorithm performs the following steps:
     \f$f_y\f$ (ratios of 10:1 or more)), then you are probably using patternSize=cvSize(rows,cols)
     instead of using patternSize=cvSize(cols,rows) in @ref findChessboardCorners.
 
+@note
+    The function may throw exceptions, if unsupported combination of parameters is provided or
+    the system is underconstrained.
+
 @sa
    calibrateCameraRO, findChessboardCorners, solvePnP, initCameraMatrix2D, stereoCalibrate,
    undistort
```

File: doc/pattern_tools/test_charuco_board.py [hunk 6]
```diff
@@ -100,15 +105,22 @@ def test_aruco_marker_sizes(self):
...
                     from_svg_img = cv.imread(filepng)
 
                     #test
-                    _charucoCorners, _charucoIds, markerCorners_svg, markerIds_svg = charuco_detector.detectBoard(from_svg_img)
-                    _charucoCorners, _charucoIds, markerCorners_cv, markerIds_cv = charuco_detector.detectBoard(from_cv_img)
+                    _charucoCorners, _charuco_ids_svg, marker_corners_svg, marker_ids_svg = charuco_detector.detectBoard(from_svg_img)
+                    _charucoCorners, _charuco_ids_cv, marker_corners_cv, marker_ids_cv = charuco_detector.detectBoard(from_cv_img)
+                    marker_corners_svg_map, marker_corners_cv_map = {}, {}
+                    for i in range(len(marker_ids_svg)):
+                        marker_corners_svg_map[int(marker_ids_svg[i])] = marker_corners_svg[i]
```
Comment (opencv-alalek) [inline doc/pattern_tools/test_charuco_board.py:116]: @AleksandrPanov Please take a look on these changes

File: modules/calib/src/calibinit.cpp [hunk 7]
```diff
@@ -72,6 +72,7 @@
 #include "precomp.hpp"
 #include "circlesgrid.hpp"
 #include "opencv2/imgproc/imgproc_c.h"
+#include "opencv2/flann.hpp"
```
Comment (opencv-alalek) [inline modules/calib/src/calibinit.cpp:75]: BTW, there is no `_c.h` file even in 4.x: https://github.com/opencv/opencv/blob/4.9.0/modules/calib3d/src/calibinit.cpp#L74

File: modules/core/perf/perf_arithm.cpp [hunk 8]
```diff
@@ -1,6 +1,6 @@
 #include "perf_precomp.hpp"
-#include "opencv2/core/softfloat.hpp"
 #include <numeric>
+#include "opencv2/core/softfloat.hpp"
```
Comment (opencv-alalek) [inline modules/core/perf/perf_arithm.cpp:3]: Looks like unnecessary changes in this file.
As there are dedicated PRs to 4.x and 5.x - all questions to the author why do we have these differences.

File: modules/dnn/perf/perf_convolution.cpp [hunk 9]
```diff
@@ -878,18 +897,104 @@ PERF_TEST_P_(Conv, conv)
...
+    }
     SANITY_CHECK_NOTHING();
 }
 
+ConvParamGenerator conv_params(testConvolution_Configs, sizeof(testConvolution_Configs) / sizeof(testConvolution_Configs[0]));
 INSTANTIATE_TEST_CASE_P(/**/, Conv, Combine(
-    ConvParamID::all(),
-    dnnBackendsAndTargets(/* withInferenceEngine = */false, /* obsolete_withHalide = */false)  // defined in ../test/test_common.hpp
+    conv_params.all(),
+    dnnBackendsAndTargets(false, false)  // defined in ../test/test_common.hpp
```
Comment (opencv-alalek) [inline modules/dnn/perf/perf_convolution.cpp:978]: Preserve 5.x coding style:

```.diff
@@ -975,26 +975,26 @@ PERF_TEST_P_(Conv_Depthwise, conv)
 ConvParamGenerator conv_params(testConvolution_Configs, sizeof(testConvolution_Configs) / sizeof(testConvolution_Configs[0]));
 INSTANTIATE_TEST_CASE_P(/**/, Conv, Combine(
     conv_params.all(),
-    dnnBackendsAndTargets(false, false)  // defined in ../test/test_common.hpp
+    dnnBackendsAndTargets(/* withInferenceEngine = */false, /* obsolete_withHalide = */false)  // defined in ../test/test_common.hpp
 ));
 
 ConvParamGenerator conv_1x1_params(testConvolution_1x1_Configs, sizeof(testConvolution_1x1_Configs) / sizeof(testConvolution_1x1_Configs[0]));
 INSTANTIATE_TEST_CASE_P(/**/, Conv_1x1, Combine(
     conv_1x1_params.all(),
-    dnnBackendsAndTargets(false, false)  // defined in ../test/test_common.hpp
+    dnnBackendsAndTargets(/* withInferenceEngine = */false, /* obsolete_withHalide = */false)  // defined in ../test/test_common.hpp
 ));
 
 ConvParamGenerator conv_3x3S1D1_params(testConvolution_3x3S1D1_Configs, sizeof(testConvolution_3x3S1D1_Configs) / sizeof(testConvolution_3x3S1D1_Configs[0]));
 INSTANTIATE_TEST_CASE_P(/**/, Conv_3x3S1D1, Combine(
     conv_3x3S1D1_params.all(),
-    dnnBackendsAndTargets(false, false),  // defined in ../test/test_common.hpp
+    dnnBackendsAndTargets(/* withInferenceEngine = */false, /* obsolete_withHalide = */false),  // defined in ../test/test_common.hpp
     testing::Values(true, false)  // enable Winograd or not
 ));
 
 ConvParamGenerator conv_depthwise_params(testConvolution_Depthwise_Configs, sizeof(testConvolution_Depthwise_Configs) / sizeof(testConvolution_Depthwise_Configs[0]));
 INSTANTIATE_TEST_CASE_P(/**/, Conv_Depthwise, Combine(
     conv_depthwise_params.all(),
-    dnnBackendsAndTargets(false, false)  // defined in ../test/test_common.hpp
+    dnnBackendsAndTargets(/* withInferenceEngine = */false, /* obsolete_withHalide = */false)  // defined in ../test/test_common.hpp
 ));
 
 } // namespace
```

File: modules/dnn/perf/perf_layer.cpp [hunk 10]
```diff
@@ -683,7 +683,119 @@ PERF_TEST_P_(Layer_GatherElements, GatherElements)
...
+
+        SANITY_CHECK_NOTHING();
+    }
+};
+
+PERF_TEST_P_(Layer_Attention, VisionTransformer) {
+    test_layer({1, 197, 768}, {768, 768, 768}, 12);
+}
+
+INSTANTIATE_TEST_CASE_P(/**/, Layer_Slice, dnnBackendsAndTargets(false, false));
```
Comment (opencv-alalek) [inline modules/dnn/perf/perf_layer.cpp:798]: Don't change exited 5.x code:

```.diff
-INSTANTIATE_TEST_CASE_P(/**/, Layer_Slice, dnnBackendsAndTargets(false, false));
+INSTANTIATE_TEST_CASE_P(/**/, Layer_Slice, dnnBackendsAndTargets(/* withInferenceEngine = */false, /* obsolete_withHalide = */false));
```

File: modules/dnn/perf/perf_net.cpp [hunk 11]
```diff
@@ -111,6 +110,7 @@ PERF_TEST_P_(DNNTestNetwork, Inception_5h)
 
 PERF_TEST_P_(DNNTestNetwork, SSD)
 {
+    applyTestTag(CV_TEST_TAG_DEBUG_VERYLONG);
```
Comment (opencv-alalek) [inline modules/dnn/perf/perf_net.cpp:113]: empty line after that https://github.com/opencv/opencv/blob/4.9.0/modules/dnn/perf/perf_net.cpp#L144

to avoid merge conflicts in the future:

File: modules/dnn/perf/perf_net.cpp [hunk 12]
```diff
@@ -155,7 +157,10 @@ PERF_TEST_P_(DNNTestNetwork, Inception_v2_SSD_TensorFlow)
```
Comment (opencv-alalek) [inline modules/dnn/perf/perf_net.cpp:157]: Lost changes for `Inception_v2_SSD_TensorFlow`:

https://github.com/opencv/opencv/blame/4.9.0/modules/dnn/perf/perf_net.cpp#L216

File: modules/dnn/perf/perf_net.cpp [hunk 13]
```diff
@@ -324,10 +346,9 @@ PERF_TEST_P_(DNNTestNetwork, ViTTrack) {
     randu(inp2, 0.0f, 1.0f);
     inp1 = blobFromImage(inp1, 1.0, Size(), Scalar(), false);
     inp2 = blobFromImage(inp2, 1.0, Size(), Scalar(), false);
-    processNet("", "dnn/onnx/models/vitTracker.onnx", {std::make_tuple(inp1, "template"), std::make_tuple(inp2, "search")});
+    processNet("", "dnn/onnx/models/object_tracking_vittrack_2023sep.onnx", {std::make_tuple(inp1, "template"), std::make_tuple(inp2, "search")});
```
Comment (opencv-alalek) [inline modules/dnn/perf/perf_net.cpp:349]: Need to change test name too: https://github.com/opencv/opencv/blame/4.9.0/modules/dnn/perf/perf_net.cpp#L419-L419C41

File: modules/dnn/perf/perf_net.cpp [hunk 14]
```diff
@@ -339,6 +360,13 @@ PERF_TEST_P_(DNNTestNetwork, EfficientDet_int8)
...
 }
 
+PERF_TEST_P_(DNNTestNetwork, VIT_B_32)
+{
+    applyTestTag(CV_TEST_TAG_DEBUG_VERYLONG);
+
+    processNet("", "dnn/onnx/models/vit_b_32.onnx", cv::Size(224, 224));
+}
+
 INSTANTIATE_TEST_CASE_P(/*nothing*/, DNNTestNetwork, dnnBackendsAndTargets());
```
Comment (opencv-alalek) [inline modules/dnn/perf/perf_net.cpp:371]: File changes mismatch:

```.diff
@@ -111,6 +111,7 @@ PERF_TEST_P_(DNNTestNetwork, Inception_5h)
 PERF_TEST_P_(DNNTestNetwork, SSD)
 {
     applyTestTag(CV_TEST_TAG_DEBUG_VERYLONG);
+
     processNet("dnn/VGG_ILSVRC2016_SSD_300x300_iter_440000.caffemodel", "dnn/ssd_vgg16.prototxt", cv::Size(300, 300));
 }
 
@@ -152,6 +153,8 @@ PERF_TEST_P_(DNNTestNetwork, opencv_face_detector)
 
 PERF_TEST_P_(DNNTestNetwork, Inception_v2_SSD_TensorFlow)
 {
+    applyTestTag(CV_TEST_TAG_DEBUG_VERYLONG);
+
     processNet("dnn/ssd_inception_v2_coco_2017_11_17.pb", "ssd_inception_v2_coco_2017_11_17.pbtxt", cv::Size(300, 300));
 }
 
@@ -339,7 +342,7 @@ PERF_TEST_P_(DNNTestNetwork, CRNN) {
     processNet("", "dnn/text_recognition_CRNN_EN_2021sep.onnx", inp);
 }
 
-PERF_TEST_P_(DNNTestNetwork, ViTTrack) {
+PERF_TEST_P_(DNNTestNetwork, VitTrack) {
     Mat inp1(cv::Size(128, 128), CV_32FC3);
     Mat inp2(cv::Size(256, 256), CV_32FC3);
     randu(inp1, 0.0f, 1.0f);
```

File: samples/dnn/segmentation.py [hunk 15]
```diff
@@ -5,7 +5,7 @@
 
 from common import *
 
-backends = (cv.dnn.DNN_BACKEND_DEFAULT, cv.dnn.DNN_BACKEND_INFERENCE_ENGINE, cv.dnn.DNN_BACKEND_OPENCV,
+backends = (cv.dnn.DNN_BACKEND_DEFAULT, cv.dnn.DNN_BACKEND_HALIDE, cv.dnn.DNN_BACKEND_INFERENCE_ENGINE, cv.dnn.DNN_BACKEND_OPENCV,
```
Comment (opencv-alalek) [inline samples/dnn/segmentation.py:8]: Wrong

**Comment by opencv-alalek** [comment]
Need to tune failed OpenCL tests: http://pullrequest.opencv.org/buildbot/builders/precommit_opencl_linux/builds/100345

**Comment by opencv-alalek** [comment]
MSVS warnings: http://pullrequest.opencv.org/buildbot/builders/precommit_opencl/builds/100315

**Comment by asmorkalov** [comment]
MSVS warnings -  they existed before merge. Will fix with the next PR.
Need to tune failed OpenCL tests. The test related to newly added model instead old Torch. Let's fix with dedicated PR.

**Review by opencv-alalek — APPROVED** [review]
